### PR TITLE
[0.73] Merge to RN 0.73.4

### PR DESCRIPTION
**Pull request by Saadnajmi**
Merge to 0.73.4 upstream

### CODE REVIEW

File: packages/react-native/React/CoreModules/RCTDeviceInfo.mm [hunk 1]
```diff
@@ -84,11 +84,44 @@ - (void)initialize
...
+  [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationDidBecomeActiveNotification object:nil];
+
+  [[NSNotificationCenter defaultCenter] removeObserver:self name:RCTUserInterfaceStyleDidChangeNotification object:nil];
+
+  [[NSNotificationCenter defaultCenter] removeObserver:self name:RCTWindowFrameDidChangeNotification object:nil];
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(invalidate)
+                                               name:RCTBridgeWillInvalidateModulesNotification
+                                             object:nil];
```
Comment (Saadnajmi) [inline packages/react-native/React/CoreModules/RCTDeviceInfo.mm:124]: Some of these may be ifdefed out for macOS

File: packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm [hunk 2]
```diff
@@ -40,6 +42,50 @@
...
+  // A bundle ID uniquely identifies a single app throughout the system. [Source: Apple docs]
+  NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
+
+  // An alphanumeric string that uniquely identifies a device to the app's vendor. [Source: Apple docs]
+  NSString *identifierForVendor = [[UIDevice currentDevice] identifierForVendor].UUIDString;
+
+  NSString *rawDeviceId = [NSString stringWithFormat:@"apple-%@-%@", identifierForVendor, bundleId];
+
+  return getSHA256(rawDeviceId);
+}
```
[the remark this answers is not in the document]
Comment (Saadnajmi) [reply packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm]: Fixed

File: packages/react-native/React/CoreModules/RCTDeviceInfo.mm (hunk 1 above)
Comment (Saadnajmi) [reply packages/react-native/React/CoreModules/RCTDeviceInfo.mm]: Fixed

File: packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm [hunk 3]
```diff
@@ -78,8 +80,29 @@
   // A bundle ID uniquely identifies a single app throughout the system. [Source: Apple docs]
   NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
 
+#if !TARGET_OS_OSX // [macOS]
   // An alphanumeric string that uniquely identifies a device to the app's vendor. [Source: Apple docs]
   NSString *identifierForVendor = [[UIDevice currentDevice] identifierForVendor].UUIDString;
+#else // [macOS
```
Comment (JunielKatarn) [inline packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm:86]: (nit) missing closing bracket `]`?

**Review by JunielKatarn — APPROVED** [review]

**Review by amgleitman — APPROVED** [review]

File: packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm (hunk 3 above)
Comment (Saadnajmi) [reply packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm:86]: Nope, the closing bracket is part of the endif. See https://github.com/microsoft/react-native-macos/blob/main/docs/DiffsWithUpstream.md

**Pull request merged**